### PR TITLE
drag drop screener tests

### PR DESCRIPTION
**Pull request by arkogupta**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes
Added screener tests for the drag and drop to reorder columns in current list control feature. I have used [JS-DragAndDrop-Simulator](https://github.com/Photonios/JS-DragAndDrop-Simulator) to simulate the drag and dragover to render the drop hint. Also includes test for gripper icon and darkening of column header on hovering.

#### Focus areas to test

(optional)

### CODE REVIEW

File: apps/vr-tests/src/stories/DetailsHeader.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    />
+  ))
+
+
+storiesOf('DetailsHeader', module)
+  .addDecorator(FabricDecorator)
+  .addDecorator(story => (
+    <Screener
+      steps={new Screener.Steps()
+        .snapshot('default1', { cropTo: '.testWrapper' })
```
Comment (micahgodbolt) [inline apps/vr-tests/src/stories/DetailsHeader.stories.tsx]: appears to be 3 snaps of the same selector

File: apps/vr-tests/src/stories/DetailsHeader.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+const _columnReorderProps = {
+  frozenColumnCountFromStart: 1,
+  frozenColumnCountFromEnd: 1,
+  handleColumnReorder: this._dummyFunction
+};
+
+
+
+storiesOf('DetailsHeader', module)
```
Comment (micahgodbolt) [inline apps/vr-tests/src/stories/DetailsHeader.stories.tsx]: Does this storiesOf string need to be unique? maybe it's the 'add' function. But wanted to make sure this wasn't causing any problems.

**Comment by arkogupta** [comment]
Pardon me, I created the PR to view the screener snapshots which were getting created, to validate the code I was writing and test a few scenarios. Should have left a comment to mention the same.
Hence the code here is not final and is in a very unpolished and incomplete state. Will make the final changes soon. 

Thanks
Arko Gupta

**Comment by micahgodbolt** [comment]
@arkogupta makes sense! You can also run visual testing locally by cd'ing into apps/vr-tests and running `npm run screener:local`

**Comment by arkogupta** [comment]
@micahgodbolt thanks a lot! Even Kevin told me about it. Wasn't aware of that earlier.

**Comment by arkogupta** [comment]
> Hey @arkogupta, it looks like there's room added to both the top and bottom of the column headers. I have two questions:
> 
> 1. Will that affect the layout of Detailslist and if not
> 2. Can we then put the caret and lollipop above the headers as originally spec'd?

Hi @betrue-final-final ,
Correct me if I am wrong but I think the room which you're talking about is the padding added due to the "test-wrapper" div in storyBook. The snapshots can be taken without the spacing as well, by omitting out the "{ cropTo: '.testWrapper' }" parameter in the snapshot API. Tell me if that needs to be done, then I'll make changes accordingly.

**Comment by betrue-final-final** [comment]
It would be great if you could update that.

**Comment by arkogupta** [comment]
@betrue-final-final Updated !
@micahgodbolt @KevinTCoughlin Could you please review ?

**Comment by betrue-final-final** [comment]
It still looks larger. Just trying to understand the change.

**Comment by arkogupta** [comment]
I think it's appearing bigger because the column headers are single letter names. I don't think there's any reason for the size to change. The borders introduced due to the test wrapper are now cropped out.

File: apps/vr-tests/screener.config.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ module.exports = {
   apiKey: process.env.SCREENER_API_KEY,
   resolution: '1024x768',
   baseBranch: (process.env.TRAVIS_PULL_REQUEST !== 'false' && process.env.TRAVIS_BRANCH) || 'master',
-  failureExitCode: 0
+  failureExitCode: 0,
+  cssAnimations: true
```
Comment (micahgodbolt) [inline apps/vr-tests/screener.config.js]: Is this change required? We're seeing changes in components that use animation (like spinner and teaching bubble). With animation set to true we might see these components flicker all the time. Can this be turned on for a single test rather than all of them?

Comment (arkogupta) [reply apps/vr-tests/screener.config.js]: If I pass 'cssAnimations' as true then I don't get the border while dragging in the snapshot, because it fades away. But on passing it as false (or not specifying it) the border persists and as a result the drop hint and the border appear together. So although having the border makes the screener test more robust, but functionally is a bit misleading. So shall I keep the config file unchanged( i.e animation set to false)
The drop hint and the border can be shown independently in different snapshots if we can have different value of cssAnimations in the config file for the 2 snaphots.
I am not sure how/if we can modify configs for a single test/different steps within the same test.

Comment (micahgodbolt) [reply apps/vr-tests/screener.config.js]: screener team is looking at implementing settings per story (so that you can turn animations on when you need them). Please remove cssAnimations for now, and we'll turn i back on once we can do it for just this one test.

Comment (arkogupta) [reply apps/vr-tests/screener.config.js]: Alright. Removed.

**Comment by arkogupta** [comment]
@KevinTCoughlin Is it possible to merge now ?

**Comment by KevinTCoughlin** [comment]
> 
> 
> @KevinTCoughlin Is it possible to merge now ?

We'll just need @micahgodbolt to approve or an admin to reset his review. I'd prefer he approved it himself once back from OOF. Thank you for keeping up with this and getting all the checks to pass / be approved 👍.

**Comment by KevinTCoughlin** [comment]
@arkogupta how time sensitive is this given that it is updating tests?

I'd like an admin to approve since we're pulling in a lib as script for your tests and I haven't dealt with it in this repo before.

**Comment by arkogupta** [comment]
> @arkogupta how time sensitive is this given that it is updating tests?
> 
> I'd like an admin to approve since we're pulling in a lib as script for your tests and I haven't dealt with it in this repo before.


Yeah I agree with that. Even I was sceptical in using a third party script, but it's making it feasible to simulate drag-drop which otherwise wouldn't have been possible. We should absolutely wait for Micah's or someone else's approval on this. Thanks :)

**Comment by micahgodbolt** [comment]
Looks like we're close with that new screener feature. we just need to update the version of screen to take advantage of it. @arkogupta can you bump the package version in apps/vr-tests/package.json and run a `npm run generate` to create new snapshots? This will add the new screener package to the repo and give you access to that command.

File: package.json [hunk 4]
```diff
@@ -50,5 +50,8 @@
       "node ./scripts/lint-staged/prettier",
       "git add"
     ]
+  },
+  "dependencies": {
```
Comment (micahgodbolt) [inline package.json]: not necessary in this package.

**Comment by micahgodbolt** [comment]
Restarting build. With screener updates we should be good to go

**Comment by arkogupta** [comment]
@betrue-final-final @KevinTCoughlin I think now this is in a good state to be merged.

**Comment by micahgodbolt** [comment]
@arkogupta thanks for your patience! I'm glad we were able to take the time to add these tests in the correct way (and get some new features added to screener to boot).
